common/include/common/time.h

```diff
@@ -61,8 +61,8 @@ static inline void nsleep(uint64_t ns)
 #else
   const struct timespec ts =
   {
-    .tv_sec  = ns / 1e9,
-    .tv_nsec = ns - ((ns / 1e9) * 1e9)
+    .tv_sec  = ns / (time_t)1000000000L,
+    .tv_nsec = ns - ((ns / 1000000000L) * 1000000000L)
```

Comment:
This generates identical code on all optimization levels ([subtraction](https://godbolt.org/z/v8qs8Pqhh) vs [modulo](https://godbolt.org/z/h8EPh45Wo)), so why don't we just use `%`? 
```suggestion
    .tv_nsec = ns % 1000000000,
```